crates/wasm-encoder/src/reencode.rs

```diff
@@ -0,0 +1,1308 @@
+                    reencoder.parse_export_section(&mut exports, section)?;
+                    module_section_flush_codes(module, &exports, &mut codes);
+                }
+                wasmparser::Payload::StartSection { func, .. } => {
+                    module_section_flush_codes(
+                        module,
+                        &crate::StartSection {
+                            function_index: reencoder.function_index(func),
+                        },
+                        &mut codes,
+                    );
+                }
+                wasmparser::Payload::ElementSection(section) => {
+                    let mut elements = crate::ElementSection::new();
+                    reencoder.parse_element_section(&mut elements, section)?;
+                    module_section_flush_codes(module, &elements, &mut codes);
+                }
+                wasmparser::Payload::DataCountSection { .. } => (),
+                wasmparser::Payload::DataSection(section) => {
+                    let mut data = crate::DataSection::new();
+                    reencoder.parse_data_section(&mut data, section)?;
+                    module_section_flush_codes(module, &data, &mut codes);
+                }
+                wasmparser::Payload::CodeSectionStart { .. } => {
+                    if let Some(codes) = codes.replace(crate::CodeSection::new()) {
+                        module.section(&codes);
+                    }
+                }
+                wasmparser::Payload::CodeSectionEntry(section) => {
+                    let codes = codes.get_or_insert_with(crate::CodeSection::new);
+                    reencoder.parse_function_body(codes, section)?;
+                }
+                wasmparser::Payload::ModuleSection { .. }
+                | wasmparser::Payload::InstanceSection(_)
+                | wasmparser::Payload::CoreTypeSection(_)
+                | wasmparser::Payload::ComponentSection { .. }
+                | wasmparser::Payload::ComponentInstanceSection(_)
+                | wasmparser::Payload::ComponentAliasSection(_)
+                | wasmparser::Payload::ComponentTypeSection(_)
+                | wasmparser::Payload::ComponentCanonicalSection(_)
+                | wasmparser::Payload::ComponentStartSection { .. }
+                | wasmparser::Payload::ComponentImportSection(_)
+                | wasmparser::Payload::ComponentExportSection(_) => {
+                    unreachable!("wasm component model section is not allowed in core wasm module")
+                }
+                wasmparser::Payload::CustomSection(section) => {
+                    module_section_flush_codes(
+                        module,
+                        &reencoder.custom_section(section),
+                        &mut codes,
+                    );
+                }
+                wasmparser::Payload::UnknownSection { id, .. } => {
+                    unreachable!("unknown wasm section with id {id}")
```

Comment:
Could this transfer the section as-is without parsing instead of panicking? (not that it's too useful but removing panics seems nice)

Reply:
Done